src/core/stdcpp/string.d

```diff
@@ -0,0 +1,337 @@
+﻿/**
+ * D header file for interaction with C++ std::string.
+ *
+ * Copyright: Copyright Guillaume Chatelet 2014 - 2015.
+ * License: Distributed under the
+ *      $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost Software License 1.0).
+ *    (See accompanying file LICENSE)
+ * Authors:   Guillaume Chatelet
+ * Source:    $(DRUNTIMESRC core/stdcpp/string.d)
+ */
+
+module core.stdcpp.string;
+
+/**
+ * Because of broken name mangling of extern C++, this file is only available
+ * for Linux platform right now. Name mangling is hardcoded with pragmas.
+ */
+version(linux):
+pragma(lib, "stdc++");
+
+///////////////////////////////////////////////////////////////////////////////
+// std::string declaration.
+//
+// Current caveats :
+// - manual name mangling (=> only string is functionnal, wstring won't work)
+//   https://issues.dlang.org/show_bug.cgi?id=14086
+//   https://issues.dlang.org/show_bug.cgi?id=14178
+// - won't work with custom allocators.
+// - iterators are implemented as pointers
+// - no reverse_iterator nor rbegin/rend
+// - missing functions : replace, swap
+///////////////////////////////////////////////////////////////////////////////
+
+import core.stdcpp.allocator;
+
+extern(C++, std):
+
+/**
+ * Character traits classes specify character properties and provide specific
+ * semantics for certain operations on characters and sequences of characters.
+ */
+struct char_traits(CharT) {}
+
+/**
+ * The basic_string is the generalization of class string for any character
+ * type.
+ */
+struct basic_string(T, TRAITS = char_traits!T, ALLOC = allocator!T)
+{
+    enum size_t npos = size_t.max;
+
+    alias value_type = T;
+    alias traits_type = TRAITS;
+    alias allocator_type = ALLOC;
+    alias reference = ref T;
+    alias const_reference = ref const(T);
+    alias pointer = T*;
+    alias const_pointer = const(T*);
+    alias iterator = pointer;
+    alias const_iterator = const_pointer;
+    // alias reverse_iterator
+    // alias const_reverse_iterator
+    alias difference_type = ptrdiff_t;
+    alias size_type = size_t;
+
+    // Ctor/dtor
+    pragma(mangle, "_ZNSsC1Ev")
```

Comment:
I believe you've already mentioned this, but all these ``pragma(mangle)`` declarations need to be removed.  The compiler should mangle this for us, and if it doesn't, then raise a new issue.  Someone will need to check compatibility with Windows too, which uses a completely different mangling ABI.